bsoup/bsoup.go

```diff
@@ -0,0 +1,346 @@
+package bsoup
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"sort"
+	"strconv"
+
+	"github.com/dustmop/soup"
+	"go.starlark.net/starlark"
+	"golang.org/x/net/html"
+)
+
+// runScriptContent runs the script, and return the global value of `resultName` as a string
+func runScriptContent(scriptSource, resultName string) (string, error) {
+	thread := &starlark.Thread{}
+
+	environment := make(map[string]starlark.Value)
+	defineBuiltins(environment)
+
+	outEnv, err := starlark.ExecFile(thread, "", scriptSource, environment)
+	if err != nil {
+		return "", err
+	}
+
+	resultVal := outEnv[resultName]
+	if err != nil {
+		return "", err
+	}
+
+	return resultVal.String(), nil
+}
+
+func defineBuiltins(environment map[string]starlark.Value) {
+	environment["OpenHtml"] = starlark.NewBuiltin("OpenHtml", OpenHTML)
+}
+
+// OpenHTML parses html from a provided filename, and returns it as a SoupNode
+func OpenHTML(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var filename starlark.String
+	if err := starlark.UnpackArgs("OpenHtml", args, kwargs, "filename", &filename); err != nil {
+		return nil, err
+	}
+
+	filenameText, err := AsString(filename)
+	if err != nil {
+		return nil, err
+	}
+
+	content, err := ioutil.ReadFile(filenameText)
```

Comment:
Actually, I think this is totally wrong. It's just how I originally coded it, but you're right that we should be parsing strings instead of accessing the file system.